rai-sdk/src/main/java/com/relationalai/TransactionAsyncResult.java

```diff
@@ -4,15 +4,32 @@
 
 public class TransactionAsyncResult extends Entity {
 
+    public Boolean gotCompletedResult;
```

Comment:
let's call this `gotCompleteResult`, not `gotCompletedResult`, inline with the Go SDK: https://github.com/RelationalAI/rai-sdk-go/blob/bbc1c3bba86b43865e983d992d48161e5652530f/rai/models.go#L302-L303

It doesn't mean that the transaction was completed, it means that it was either completed or aborted, and we got the results.

Reply:
I guess there are a few cases we're covering with this type:
1. `ExecuteAsync` => fast path; contains results. state is `aborted` or `completed`
2. `ExecuteAsync` => slow path; doesn't contain results. state is `running`
3. `Execute` => regardless of path, contains results. either `aborted` or `completed`

I guess `gotCompleteResult` differentiates (1) from (2)? But why do we need it? Can't we use the state to differentiate?

What am I missing @NHDaly

Reply:
We discussed how in the original spec (before you joined the project, i think, @vilterp), we inteded to have a fourth state:
4. ExecuteAsync => fast path; does not contain results (because they are too big to send over the wire, and you presumably want to page them). state is aborted or completed

Reply:
K thanks. I forgot that it was future proofing and not a case we need now.